ts/torch_handler/base_handler.py

```diff
@@ -78,38 +81,101 @@ def initialize(self, context):
+            except ImportError as error:
+                onnx_enabled = False
+                logger.warning("proceeding without onnxruntime")
 
         # model def file
         model_file = self.manifest["model"].get("modelFile", "")
 
         if model_file:
             logger.debug("Loading eager model")
-            self.model = self._load_pickled_model(model_dir, model_file, model_pt_path)
+            self.model = self._load_pickled_model(
+                model_dir, model_file, self.model_pt_path
+            )
             self.model.to(self.device)
-        else:
-            logger.debug("Loading torchscript model")
-            if not os.path.isfile(model_pt_path):
-                raise RuntimeError("Missing the model.pt file")
+            self.model.eval()
+
+        # Convert your model by following instructions: https://pytorch.org/tutorials/intermediate/nvfuser_intro_tutorial.html
+        # For TensorRT support follow instructions here: https://pytorch.org/TensorRT/getting_started/getting_started_with_python_api.html#getting-started-with-python-api
+        elif self.model_pt_path.endswith(".pt"):
+            self.model = self._load_torchscript_model(self.model_pt_path)
+            self.model.eval()
+
+        # Convert your model by following instructions: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html
+        elif self.model_pt_path.endswith(".onnx") and onnx_enabled:
+            # self.model = self._load_onnx_model(self.model_pt_path)
+            providers = (
+                ["CUDAExecutionProvider", "CPUExecutionProvider"]
+                if self.map_location == "cuda"
+                else ["CPUExecutionProvider"]
+            )
 
-            self.model = self._load_torchscript_model(model_pt_path)
+            # Set the right inference options, we can add more options here depending on what people want
+            sess_options = ort.SessionOptions()
+            # sess_options.intra_op_num_threads = psutil.cpu_count(logical=True)
+
+            # Start an inference session
+            ort_session = ort.InferenceSession(
+                self.model_pt_path, providers=providers, sess_options=sess_options
+            )
+            self.model = ort_session
+
+        else:
+            raise RuntimeError("No model weights could be loaded")
 
-        self.model.eval()
         if ipex_enabled:
             self.model = self.model.to(memory_format=torch.channels_last)
             self.model = ipex.optimize(self.model)
 
-        logger.debug("Model file %s loaded successfully", model_pt_path)
+        logger.debug("Model file %s loaded successfully", self.model_pt_path)
 
         # Load class mapping for classifiers
         mapping_file_path = os.path.join(model_dir, "index_to_name.json")
         self.mapping = load_label_mapping(mapping_file_path)
 
         self.initialized = True
 
+    def _load_onnx_model(self, model_onnx_path):
```

Comment:
are we going to replace the above Lines 120-137 with this function?

Reply:
Yeah this is a typo, will fix